src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java

```diff
@@ -160,4 +163,17 @@ public void testVersionExists() throws IOException {
       assertReachable(repo);
     }
   }
+  
+  @Test
+  public void testOrder() throws IOException {
+    List<String> names = new ArrayList<>();
```

Comment:
Sorry, I'm on a little bit of a java8 stuff kick, but can we just check sorted vs unsorted?

```java
    List<String> namesLowerCase = apis.stream().map(api -> api.getString("name").toLowerCase(Locale.US)).collect(Collectors.toList());

    Assert.assertTrue(Comparators.isInOrder(namesLowerCase, String::compareTo));
    //OR
    Assert.assertEquals(namesLowerCase.stream().sorted().collect(Collectors.toList()), namesLowerCase) // if you want to see differences?
```

Reply:
Hmm... whatever way it is, I think the final comparison of `.compareTo()` is simply based on the Unicode value, so it may not work as intended even if the strings were lower-cased with a proper locale? Should we be really bothered with a locale in this case?

Reply:
Yes, lower casing always needs a locale. Even basic ASCII letters such as i and I change case differently in different locales. In this case, all names are English and ASCII so Locale. US works. 

Let me play with Java 8.

Reply:
The Java 8 suggestions work well as long as the test doesn't fail. However when it does fail (and this did fail initially, and will likely fail again when we add libraries) the more detailed error message from the individual non-streamed checks makes the specific error more apparent. 

This test is what made me realize lower casing was needed. Otherwise Datastore sorted after DNS, to my surprise.